rfd/0008-api.md

```diff
@@ -0,0 +1,241 @@
+---
+authors: Alexander Klizhentas (sasha@gravitational.com)
+state: discussion
+---
+
+# RFD 8 - API and Client Libraries
+
+## What
+
+Document and publish API OpenAPI spec and documentation.
+Unify Golang Client experience.
+
+## Why
+
+There are several problems with Go client libraries:
+
+* Go library example pulls entire teleport dependency including `lib/auth` and `lib/services`.
```

Comment:
Note to self: Also, constants like `RoleProxy` are in the teleport package, adding another dependency.